showcase-v1alpha2/src/v1alpha2/echo_client.ts

```diff
@@ -0,0 +1,688 @@
+   * The fourth parameter keeps the raw response object of an object
+   * representing [PaginationResponse]{@link
+   * google.showcase.v1alpha2.PaginationResponse}.
+   * @returns {Promise} - The promise which resolves to an array.
+   *   The first element of the array is Array of number.
+   *
+   *   When autoPaginate: false is specified through options, the array has
+   * three elements. The first element is Array of number in a single response.
+   *   The second element is the next request object if the response
+   *   indicates the next page exists, or null. The third element is
+   *   an object representing [PaginationResponse]{@link
+   * google.showcase.v1alpha2.PaginationResponse}.
+   *
+   *   The promise has a method named "cancel" which cancels the ongoing API
+   * call.
+   *
+   * @example
+   *
+   * const showcase = require('showcase.v1alpha2');
+   *
+   * const client = new showcase.v1alpha2.EchoClient({
+   *   // optional auth parameters.
+   * });
+   *
+   * // Iterate over all elements.
+   * const maxResponse = 0;
+   *
+   * client.pagination({maxResponse: maxResponse})
+   *   .then(responses => {
+   *     const resources = responses[0];
+   *     for (let i = 0; i < resources.length; i += 1) {
+   *       // doThingsWith(resources[i])
+   *     }
+   *   })
+   *   .catch(err => {
+   *     console.error(err);
+   *   });
+   *
+   * // Or obtain the paged response.
+   * const maxResponse = 0;
+   *
+   *
+   * const options = {autoPaginate: false};
+   * const callback = responses => {
+   *   // The actual resources in a response.
+   *   const resources = responses[0];
+   *   // The next request if the response shows that there are more responses.
+   *   const nextRequest = responses[1];
+   *   // The actual response object, if necessary.
+   *   // const rawResponse = responses[2];
+   *   for (let i = 0; i < resources.length; i += 1) {
+   *     // doThingsWith(resources[i]);
+   *   }
+   *   if (nextRequest) {
+   *     // Fetch the next page.
+   *     return client.pagination(nextRequest, options).then(callback);
+   *   }
+   * }
+   * client.pagination({maxResponse: maxResponse}, options)
+   *   .then(callback)
+   *   .catch(err => {
+   *     console.error(err);
+   *   });
+   */
+  pagination(
+      request: PaginationRequest, options?: gax.CallOptions,
+      callback?: GapicPaginationCallback<
+          PaginationRequest, PaginationResponse, number>):
+      Promise<GapicPaginationResponse<
+          PaginationRequest, PaginationResponse, number>> {
+    options = options || {};
+    // Here we receive an untyped array from gax and convert it to typed object.
+    return new Promise((resolve, reject) => {
+      this._innerApiCalls.pagination(request, options, callback)
+          .then((resultArray: Array<{}>) => {
+            const result: GapicPaginationResponse<
+                PaginationRequest, PaginationResponse, number> = {
+              values: resultArray[0] as number[],
```

Comment:
This stuff is basically just telling `tsc` that you know what you are doing?

Reply:
Yes. From Gax I'm getting an array with three elements, all of different types - just a disaster for the typings. I'd like to change it there, but then we'll have compatibility problems with existing gax users.

https://github.com/googleapis/gax-nodejs/blob/9a35ccf7d8c4d331889b85008e6fdca05169983c/src/api_callable.ts#L164-L172